lib/ansible/executor/task_queue_manager.py

```diff
@@ -163,6 +167,13 @@ def __init__(self, inventory, variable_manager, loader, passwords, stdout_callba
         except OSError as e:
             raise AnsibleError("Unable to use multiprocessing, this is normally caused by lack of access to /dev/shm: %s" % to_native(e))
 
+        try:
+            # Done in tqm, and not display, because this is only needed for commands that execute tasks
+            for fd in (STDIN_FILENO, STDOUT_FILENO, STDERR_FILENO):
+                os.set_inheritable(fd, False)
+        except Exception as ex:
+            self.warning(f"failed to set stdio as non inheritable: {ex}")
```

Comment:
Is there a known scenario where `os.set_inheritable` would fail or is this just covered its bases?

Reply:
Just covering the bases.  We do this type of thing in `Display` and I had originally put it there, but ran into issues like in the galaxy CLI where we rely on inheriting stdio for output.  It's likely not required, and not sure of a scenario where it would fail, but I guess just being safe.